spec/lib/models/revocable_spec.rb

```diff
@@ -33,4 +33,26 @@
       expect(subject).not_to be_revoked
     end
   end
+
+  describe :revoke_previous_refresh_token! do
+    subject { FactoryGirl.build(:access_token, previous_refresh_token: 'old_refresh_token') }
+    previous_token = FactoryGirl.build(:access_token, token: "previous_token")
+    after { subject.revoke_previous_refresh_token! }
+
+    it 'revokes the previous token if present and sets the attribute :previous_refresh_token to ""' do
+      expect(Doorkeeper::AccessToken).to receive(:by_refresh_token).with(subject.previous_refresh_token).and_return(previous_token)
```

Comment:
Revocable is a module, and as such we can only test it through a concrete class like `AccessToken`. When we stub it like this, we are [stubbing the system under test](https://robots.thoughtbot.com/don-t-stub-the-system-under-test), potentially hiding bugs in the very same class we are testing.

We can also make it more readable as a "story" following the [four phase test](https://robots.thoughtbot.com/four-phase-test):   setup, exercise, verify, and teardown. That way it will read more like a story: "this is the state of the world, and then this happens, and then this is what we'd expect", without needing to jump from line to line parsing the code in our heads.

These two comments might be a departure from current doorkeeper style of specs, and thus might look a little different than other parts of this file and other files too. I'm fine with that, as I want to move towards more readable specs.

Example for the first spec (might unveil bugs in your current branch):

```ruby
describe :revoke_previous_refresh_token! do
  it "revokes the previous token if existing, and resets the `previous_refresh_token` attribute" do
    previous_token = FactoryGirl.create(:access_token, refresh_token: "refresh_token")
    current_token = FactoryGirl.create(
      :access_token,
      previous_refresh_token: previous_token.refresh_token
    )

    current_token.revoke_previous_refresh_token!

    expect(current_token.previous_refresh_token).to be_empty
    expect(previous_token.reload).to be_revoked
  end
end
```